circuit-lib/circuitlib-rs/src/groth16_solana_verifier.rs

```diff
@@ -0,0 +1,133 @@
+use std::{collections::HashMap, fs::File, ops::Neg};
+
+use ark_bn254::Bn254;
+use ark_circom::{circom::Inputs, read_zkey};
+use ark_groth16::Proof as ArkGroth16Proof;
+use ark_serialize::{CanonicalSerialize, Compress};
+use groth16_solana::{
+    decompression::{decompress_g1, decompress_g2},
+    groth16::{Groth16Verifier, Groth16Verifyingkey},
+};
+
+use crate::{
+    arkworks_prover::{prove, ArkProof, ArkProvingKey},
+    errors::CircuitsError,
+    helpers::{change_endianness, convert_endianness_128},
+    merkle_proof_inputs::{public_inputs, MerkleTreeInfo, MerkleTreeProofInput, ProofInputs},
+};
+
+pub struct ProofCompressed {
+    pub a: [u8; 32],
+    pub b: [u8; 64],
+    pub c: [u8; 32],
+}
+
+impl ProofCompressed {
+    pub fn try_decompress(&self) -> Result<Proof, CircuitsError> {
+        let proof_a = decompress_g1(&self.a)?;
+        let proof_b = decompress_g2(&self.b)?;
+        let proof_c = decompress_g1(&self.c)?;
+        Ok(Proof {
+            a: proof_a,
+            b: proof_b,
+            c: proof_c,
+        })
+    }
+}
+
+pub struct Proof {
+    pub a: [u8; 64],
+    pub b: [u8; 128],
+    pub c: [u8; 64],
+}
+
+pub struct ProofResult {
+    pub proof: ProofCompressed,
+    pub public_inputs: Vec<[u8; 32]>,
+}
+
+pub fn merkle_inclusion_proof(
+    merkle_tree_info: &MerkleTreeInfo,
+    merkle_proof_inputs: &[MerkleTreeProofInput],
+) -> Result<ProofResult, CircuitsError> {
+    let merkle_proof_inputs_len = merkle_proof_inputs.len();
+
+    let proof_inputs = ProofInputs(merkle_proof_inputs);
+    let inputs_hashmap: HashMap<String, Inputs> = proof_inputs
+        .try_into()
+        .map_err(|_| CircuitsError::ChangeEndiannessError)?;
+
+    let path = format!(
+        "test-data/merkle{}_{}/circuit.zkey",
```

Comment:
can we pass in the path?
(I think when using this in tests the path is going to be different, so this won't work.)